src/zelos/feeds.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (C) 2020 Zeropoint Dynamics
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+# ======================================================================
+
+"""Implementation of feeds.
+
+Feeds are a way to subscribe to information that is retrieved from a
+dynamic execution of a binary, while respecting the performance
+optimizations that have been requested by the user. This way, a user can
+specify what kinds of information should be collected (each with their
+own performance cost) in a global way, without having to configure
+multiple plugins separately.
+
+To start with, there are different levels of feeds, each increasing in
+amount of verbosity as well as performance cost::
+
+                     -> Verbosity ->
+            None -> Syscalls -> Apis -> Instructions
+                    <- Performance <-
+
+The feed level determines what feeds are supplied with information. All
+feeds that are more verbose than the feed level are not provided data,
+and will not run subscribed callbacks.
+
+Example:
+    When the feed level is FeedLevel.API, subscribers to the syscall and
+    api feeds will be run, but no calls to inst feed subscribers will be
+    made.
+
+Feeds are made more powerful by command line arguments that provide ways
+to modify the feed level based on events and conditions. This allows the
+user to specify that only instructions between certain regions should be
+collected. The conditions are specified through the
+--(stop|syscall|api|inst)_feed command line flags.
+
+Example:
+    To trigger instructions to be printed only after the 'recv' syscall
```

Comment:
It would be worthwhile to greatly expand on examples in a tutorial.

Reply:
Adding to this, which I should have mentioned in #64 , a feeds tutorial should also include an introduction to ZML.